src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -153,7 +153,7 @@ public static IEnumerable<CustomAttributeInfo> GetNonInheritableAttributes(this
 #else
 				var attributeType = attribute.AttributeType;
 #endif
-				if (ShouldSkipAttributeReplication(attributeType))
+				if (ShouldSkipAttributeReplication(attributeType, ignoreInheritance: false))
```

Comment:
`ignoreInheritance: false` will result in the same behavior that was there before this parameter was added.